transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -85,11 +88,22 @@ public EmbeddedChannel(ChannelId channelId) {
      *
      * @param handlers the {@link ChannelHandler}s which will be add in the {@link ChannelPipeline}
      */
-    public EmbeddedChannel(final ChannelHandler... handlers) {
+    public EmbeddedChannel(ChannelHandler... handlers) {
         this(EmbeddedChannelId.INSTANCE, handlers);
     }
 
     /**
+     * Create a new instance with the pipeline initialized with the specified handlers.
+     *
+     * @param hasDisconnect {@code true} if this {@link Channel} will handle {@link #disconnect()} and {@link #close()}
+     *                      differently, {@code false} otherwise.
```

Comment:
@nmittler what I tried to say is that either close() and disconnect() are doing the same and so always go through `ChannelOutboundHandler.close(...)` or that they will be threaded differently and so will go through different methods in `ChannelOutboundHandler` + only close() will move the channel into inactive state. Any idea for good javadoc ?